packages/interfaces/src/filter.ts

```diff
@@ -1,17 +1,77 @@
-import type { PeerId } from "@libp2p/interface";
-
 import type { IDecodedMessage, IDecoder } from "./message.js";
-import type { ContentTopic, ThisOrThat } from "./misc.js";
-import type {
-  Callback,
-  ProtocolError,
-  SDKProtocolResult
-} from "./protocols.js";
-import type { IReceiver } from "./receiver.js";
+import type { Callback } from "./protocols.js";
+
+export type IFilter = {
+  readonly multicodec: string;
+
+  /**
+   * Subscribes to messages with specified decoders and executes callback when a message is received.
+   * In case no peers available initially - will delay subscription till connects to any peer.
+   *
+   * @param decoders - Single decoder or array of decoders to subscribe to. All decoders must share the same pubsubTopic.
+   * @param callback - Function called when a message matching the decoder's contentTopic is received.
+   * @returns Promise that resolves to true if subscription was successful, false otherwise.
```

Comment:
```suggestion
   * @returns Boolean indicating if the subscription was created successfully
```

Reply:
promise must be specified because it is the returning type